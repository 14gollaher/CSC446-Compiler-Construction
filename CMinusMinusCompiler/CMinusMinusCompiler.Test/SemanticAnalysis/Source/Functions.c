int beforeMain()
{
	return 3;
}

char main(int x, int y)
{
	return 11;
}

float afterMain(char x, int y, float z)
{
	int p;
	return 23;
}
/*This will be omitted!!!*/
