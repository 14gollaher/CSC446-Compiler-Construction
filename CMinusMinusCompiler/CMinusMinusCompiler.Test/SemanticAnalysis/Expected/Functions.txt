** Symbol Table at Depth 2 **
Lexeme          Class        Depth     Offset    Type         Size      Value(r)
--------------------------------------------------------------------------------


** Symbol Table at Depth 2 **
Lexeme          Class        Depth     Offset    Type         Size      Value(r)
--------------------------------------------------------------------------------
x               Variable     2          4         IntToken     2         -
y               Variable     2          6         IntToken     2         -


** Symbol Table at Depth 2 **
Lexeme          Class        Depth     Offset    Type         Size      Value(r)
--------------------------------------------------------------------------------
x               Variable     2          4         CharToken    1         -
y               Variable     2          5         IntToken     2         -
z               Variable     2          7         FloatToken   4         -
p               Variable     2         -2         IntToken     2         -


** Symbol Table at Depth 1 **
Lexeme          Class        Depth     Offset    Type         Size      Value(r)
--------------------------------------------------------------------------------
beforeMain      Function     1         -         IntToken     0         -
main            Function     1         -         CharToken    4         -
-               Parameter    -         -         IntToken     -         -
-               Parameter    -         -         IntToken     -         -
afterMain       Function     1         -         FloatToken   9         -
-               Parameter    -         -         CharToken    -         -
-               Parameter    -         -         IntToken     -         -
-               Parameter    -         -         FloatToken   -         -


Completed processing Functions.c
